infra/build/functions/update_build_status.py

```diff
@@ -0,0 +1,118 @@
+# Copyright 2020 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+################################################################################
+"""Cloud function to request builds."""
+import logging
+
+import google.auth
+from googleapiclient.discovery import build
+from google.cloud import ndb
+
+import build_and_run_coverage
+import build_project
+import builds_status
+from datastore_entities import BuildsHistory
+from datastore_entities import Project
+
+
+# pylint: disable=no-member
+def get_last_build(build_ids):
+  """Returns build object for the last finished build of project."""
+  credentials, image_project = google.auth.default()
+  cloudbuild = build('cloudbuild', 'v1', credentials=credentials)
+
+  for build_id in reversed(build_ids):
+    project_build = cloudbuild.projects().builds().get(projectId=image_project,
+                                                       id=build_id).execute()
+    if project_build['status'] == 'WORKING':
+      continue
+
+    if not builds_status.upload_log(build_id):
+      continue
```

Comment:
the function returns False in case the build log with the specified build id does not exist in our gcb-logs-bucket, so if the log file is missing it is better not to update it since it won't be copied over to our bucket in clusterfuzz

Reply:
Raise an exception in that case. That would be a critical error in our infrastructure.

Reply:
Alright will do